fem/bilinearform.cpp

```diff
@@ -181,7 +218,24 @@ void BilinearForm::Finalize (int skip_zeros)
 
 void BilinearForm::AddDomainIntegrator (BilinearFormIntegrator * bfi)
 {
-   dbfi.Append (bfi);
+   switch (assembly)
```

Comment:
It's a shame that we have to keep putting these switch/cases blocks in for the assembly cases.  It would be nice if we could rework some of the interface to avoid this, but it looks like we are stuck with it now for backward compatibility.

Reply:
This can go away once we have support for the FA integrators...